File: frontend/src/types/job_description.ts
export interface JobDescription {
  id: number;
  title: string;
  content: string;
  created_at: string; // ISO 8601 — from backend datetime field
}

export interface JobDescriptionCreate {
  title: string;
  content: string;
}

export interface JobDescriptionUpdate {
  title?: string;
  content?: string;
}

export interface JobDescriptionListResponse {
  items: JobDescription[];
  total: number;
}

File: backend/app/schemas/job_description.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class JobDescriptionCreate(BaseModel):
    title: str = Field(min_length=1, max_length=500)
    content: str = Field(min_length=1)


class JobDescriptionUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=500)
    content: str | None = Field(default=None, min_length=1)


class JobDescriptionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    content: str
    created_at: datetime


class JobDescriptionListResponse(BaseModel):
    items: list[JobDescriptionResponse]
    total: int
